virttest/utils_misc.py

```diff
@@ -1611,30 +1617,144 @@ def get_cpu_info(session=None):
     return cpu_info
 
 
-def yum_install(pkg_list, session=None, timeout=300):
+class RemotePackage(object):
     """
-    Try to install packages on system
+    The remote package manage class
+    """
+    def __init__(self, session, pkg):
+        """
+        :param session: sessioon object
+        :param pkg: pacakge name or list
+        """
+        if not isinstance(session, (aexpect.ShellSession, aexpect.Expect)):
+            raise exceptions.TestError("Parameters exception on session")
+        if not isinstance(pkg, list):
+            if not isinstance(pkg, str):
+                raise exceptions.TestError("pkg %s must be list or str" % pkg)
+            else:
+                self.pkg_list = [pkg, ]
+        else:
+            self.pkg_list = pkg
+        self.package_manager = None
+        self.query_cmd = None
+        self.install_cmd = None
+        self.remove_cmd = None
+        self.session = session
+
+        # Inspect and set package commands
+        for pkg_mgr in PACKAGE_MANAGERS:
+            cmd = 'which ' + pkg_mgr
+            if not self.session.cmd_status(cmd):
+                self.package_manager = pkg_mgr
+                break
+
+            if not self.package_manager:
```

Comment:
should this come out of for loop, so that it checks for all the `which pkg_mgr` and then errors out else if 1st iteration failing will raise the exception